src/node_nanomsg.cc

```diff
@@ -125,6 +130,41 @@ NAN_METHOD(Getsockopt) {
     NanReturnValue(obj);
 }
 
+NAN_METHOD(Setopt) {
+  NanScope();
+
+  int level = args[1].integer;
+  int option = args[2].integer;
+
+  if(option == NN_SOCKET_NAME){
+    utf8 str(args[3]);
+    ret(NanNew<Number>(nn_setsockopt(S, level, option, *str, str.length())));
+  } else {
+    int optval = args[3].integer;
+    ret(NanNew<Number>(nn_setsockopt(S, level, option, &optval, sizeof(optval))));
+  }
+}
+
+NAN_METHOD(Getopt) {
+  NanScope();
+
+  int optval[64];
+  int option = args[2].integer;
+  size_t optsize = sizeof(optval);
+
+  //check if the function succeeds
+  if(nn_getsockopt(S, args[1].integer, option, optval, &optsize) == 0){
```

Comment:
`rcvprio()` doesn't work because `NN_RCVPRIO` symbol is not exported

it's supposed to be `9` after `NN_SOL_SOCKET` starts at `0`

this is what's being exported:

```js
_bindings: 
   { Socket: [Function],
     Close: [Function],
     Setsockopt: [Function],
     Getsockopt: [Function],
     Bind: [Function],
     Connect: [Function],
     Shutdown: [Function],
     Send: [Function],
     Recv: [Function],
     Errno: [Function],
     PollSendSocket: [Function],
     PollReceiveSocket: [Function],
     PollStop: [Function],
     DeviceWorker: [Function],
     SymbolInfo: [Function],
     Symbol: [Function],
     Term: [Function],
     Getopt: [Function],
     Setopt: [Function],
     Err: [Function],
     NN_NS_NAMESPACE: 0,
     NN_NS_VERSION: 1,
     NN_NS_DOMAIN: 2,
     NN_NS_TRANSPORT: 3,
     NN_NS_PROTOCOL: 4,
     NN_NS_OPTION_LEVEL: 5,
     NN_NS_SOCKET_OPTION: 6,
     NN_NS_TRANSPORT_OPTION: 7,
     NN_NS_OPTION_TYPE: 8,
     NN_NS_OPTION_UNIT: 9,
     NN_NS_FLAG: 10,
     NN_NS_ERROR: 11,
     NN_NS_LIMIT: 12,
     NN_TYPE_NONE: 0,
     NN_TYPE_INT: 1,
     NN_TYPE_STR: 2,
     NN_UNIT_NONE: 0,
     NN_UNIT_BYTES: 1,
     NN_UNIT_MILLISECONDS: 2,
     NN_UNIT_PRIORITY: 3,
     NN_UNIT_BOOLEAN: 4,
     NN_VERSION_CURRENT: 2,
     NN_VERSION_REVISION: 1,
     NN_VERSION_AGE: 2,
     AF_SP: 1,
     AF_SP_RAW: 2,
     NN_INPROC: -1,
     NN_IPC: -2,
     NN_TCP: -3,
     NN_PAIR: 16,
     NN_PUB: 32,
     NN_SUB: 33,
     NN_REP: 49,
     NN_REQ: 48,
     NN_PUSH: 80,
     NN_PULL: 81,
     NN_SURVEYOR: 96,
     NN_RESPONDENT: 97,
     NN_BUS: 112,
     NN_SOCKADDR_MAX: 128,
     NN_SOL_SOCKET: 0,
     NN_LINGER: 1,
     NN_SNDBUF: 2,
     NN_RCVBUF: 3,
     NN_SNDTIMEO: 4,
     NN_RCVTIMEO: 5,
     NN_RECONNECT_IVL: 6,
     NN_RECONNECT_IVL_MAX: 7,
     NN_SNDPRIO: 8,
     NN_SNDFD: 10,
     NN_RCVFD: 11,
     NN_DOMAIN: 12,
     NN_PROTOCOL: 13,
     NN_IPV4ONLY: 14,
     NN_SOCKET_NAME: 15,
     NN_SUB_SUBSCRIBE: 1,
     NN_SUB_UNSUBSCRIBE: 2,
     NN_REQ_RESEND_IVL: 1,
     NN_SURVEYOR_DEADLINE: 1,
     NN_TCP_NODELAY: 1,
     NN_DONTWAIT: 1,
     EADDRINUSE: 48,
     EADDRNOTAVAIL: 49,
     EAFNOSUPPORT: 47,
     EAGAIN: 35,
     EBADF: 9,
     ECONNREFUSED: 61,
     EFAULT: 14,
     EFSM: 156384766,
     EINPROGRESS: 36,
     EINTR: 4,
     EINVAL: 22,
     EMFILE: 24,
     ENAMETOOLONG: 63,
     ENETDOWN: 50,
     ENOBUFS: 55,
     ENODEV: 19,
     ENOMEM: 12,
     ENOPROTOOPT: 42,
     ENOTSOCK: 38,
     ENOTSUP: 45,
     EPROTO: 100,
     EPROTONOSUPPORT: 43,
     ETERM: 156384765,
     ETIMEDOUT: 60,
     EACCES: 13,
     ECONNABORTED: 53,
     ECONNRESET: 54,
     EHOSTUNREACH: 65,
     EMSGSIZE: 40,
     ENETRESET: 52,
     ENETUNREACH: 51,
     ENOTCONN: 57,
     path: '/Users/bent/routekickr/node-nanomsg/build/Release/node_nanomsg.node' },
  Socket: 

```